flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -731,14 +731,18 @@ void draw_alarms(int x, int y, int xs, int ys, int va, int ha, int flags, int fo
 		buf[strlen(boot_reason)] = '\0';
 		pos = strlen(boot_reason);
 		buf[pos++] = ' ';
-		return;
 	}
 
 	uint8_t state;
-	int32_t len = AlarmString(&alarm, buf, sizeof(buf) - 1, blink, &state);
+	int32_t len = AlarmString(&alarm, buf + pos, sizeof(buf) - 1 - pos,
+			blink, &state);
 
 	if (len > 0) {
```

Comment:
Looks like it will be 0 if there are no warning/error/critical alarms? https://github.com/d-ronin/dRonin/blob/080150a5b271fcfb3fe36aa1e20f92fc0d956aca/flight/Libraries/alarms.c#L329-L396